artemis-protocols/artemis-mqtt-protocol/src/main/java/org/apache/activemq/artemis/core/protocol/mqtt/MQTTProtocolManagerFactory.java

```diff
@@ -64,4 +74,46 @@ public ProtocolManager createProtocolManager(ActiveMQServer server,
    public String getModuleName() {
       return MODULE_NAME;
    }
+
+
+   @Override
+   public void loadProtocolServices(ActiveMQServer server, List<ActiveMQComponent> services) {
+      services.add(new MQTTPeriodicTasks(server.getScheduledPool()));
+   }
+
+   public class MQTTPeriodicTasks implements ActiveMQComponent {
```

Comment:
extend ActiveMQScheduledComponent instead.



provide a tick method.. like other components...

Reply:
the method is actually a runnable... I have sent you a PR